src/main/java/com/exasol/validation/GitRepositoryValidator.java

```diff
@@ -25,95 +26,95 @@ public GitRepositoryValidator(final GitRepository repository) {
 
     /**
      * Validate content of a Git-based repository.
-     * 
+     *
      * @param branch name of a branch to validate on
      */
     public void validate(final String branch) {
         LOGGER.fine("Validating Git repository.");
-        final GitRepositoryContent content = this.repository.getRepositoryContent(branch);
-        final String changelog = content.getChangelogFile();
+        final GitBranchContent content = this.repository.getRepositoryContent(branch);
         final String version = content.getVersion();
-        final String changes = content.getChangesFile(version);
+        validateNewVersion(version);
+        final String changelog = content.getChangelogFile();
         validateChangelog(changelog, version);
+        final ReleaseLetter changes = content.getReleaseLetter(version);
         validateChanges(changes, version);
+    }
+
+    protected void validateNewVersion(final String newVersion) {
+        LOGGER.fine("Validating a new version.");
+        validateVersionFormat(newVersion);
         final Optional<String> latestReleaseTag = this.repository.getLatestTag();
-        validateVersion(version, latestReleaseTag);
+        latestReleaseTag.ifPresent(s -> validateNewVersionWithPreviousTag(newVersion, s));
```

Comment:
This is an awkward way to replace a regular `if`. Please use that only where you are forced too (e.g. in a stream).

Reply:
fixed